client-course-schedulizer/src/utilities/services/validation/popover.ts

```diff
@@ -33,41 +33,36 @@ NOTE: fields with default values are not check: semester and time. */
 export const addSectionSchema = object().shape({
   anticipatedSize: number()
     .typeError("global max must be a number")
-    .positive()
     .integer()
     .transform(emptyStringToNull)
     .nullable(),
   day10Used: number()
     .typeError("day 10 used must be a number")
-    .positive()
     .integer()
     .transform(emptyStringToNull)
     .nullable(),
   days: array().transform(removeUncheckedValues),
   department: string(),
   duration: number()
     .typeError("duration must be a number")
-    .positive()
     .integer()
     .transform(emptyStringToNull)
     .nullable(),
   facultyHours: number()
     .typeError("faculty hours must be a number")
-    .positive()
+    .integer()
```

Comment:
I think we don't want the integer check here. Also, I think we want to leave the positive checks? I've been able to update courses to have a negative global max for example which seems illogical.

Reply:
Positive doesn't work because it doesn't allow 0 as a possible value. I added integer because I thought it was better than nothing, but now realize that that doesn't work either since there can be fractional loads. I'll remove it, but I think we should figure out how to add a custom function to allow only positive numbers or 0.

Reply:
Can you write a .nonnegative()?

Reply:
After some research, it looks like .min(0) should do the trick

Reply:
:dab:

![](https://user-images.githubusercontent.com/33156025/110053347-3b7a4f00-7d27-11eb-8a63-acc9f5f85a4e.png)